django/forms/widgets.py

```diff
@@ -618,12 +618,13 @@ class ChoiceWidget(Widget):
     checked_attribute = {"checked": True}
     option_inherits_attrs = True
 
-    def __init__(self, attrs=None, choices=()):
+    def __init__(self, attrs=None, choices=(), option_attrs=None):
         super().__init__(attrs)
         # choices can be any iterable, but we may need to render this widget
         # multiple times. Thus, collapse it into a list so it can be consumed
         # more than once.
         self.choices = list(choices)
+        self.option_attrs = option_attrs
```

Comment:
I think this should mimic what `Widget` does in `__init__` for `attrs`:

```suggestion
        self.option_attrs = {} if option_attrs is None else option_attrs.copy()
```